### PR TITLE
#9747: Implement ttnn.tilize(_with_val_padding) Python bindings

### PR DESCRIPTION
**Pull request by yan-zaretskiy**
### Ticket
#9747 

### Problem description
Continuing the `tt_eager --> ttnn` migration plan for `tilize`.

### What's changed
Migrated Python usages of the `tilize` family of ops from deprecated `tt_lib.tensor` to `ttnn`.

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/9963189062
- [x] Model regression CI testing passes (if applicable)

### CODE REVIEW

**Review by boris-drazic — APPROVED** [review]

File: models/experimental/vgg/vgg_utils.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def format_tensor(x, target_layout, device, output_mem_config, pad_value=0.0):
                 x, device, x_padded_shape, pad_value, target_layout, output_mem_config
             )
         else:
-            return tt_lib.tensor.tilize(x, output_mem_config, use_multicore=True)
+            return ttnn.tilize(x, output_mem_config, use_multicore=True)
```
Comment (ayerofieiev-tt) [inline models/experimental/vgg/vgg_utils.py:25]: For consistency with other ops, memory_config should be a KW argument.

```suggestion
            return ttnn.tilize(x, memory_config=output_mem_config, use_multicore=True)
```

Comment (ayerofieiev-tt) [reply models/experimental/vgg/vgg_utils.py:25]: Same for `tilize_with_zero_padding`

File: ttnn/cpp/ttnn/operations/data_movement/data_movement_pybind.hpp [hunk 2]
```diff
@@ -37,13 +39,9 @@ void bind_upsample(py::module& module) {
         ttnn::upsample,
         doc,
         ttnn::pybind_arguments_t{
-            py::arg("input_tensor"),
-            py::arg("scale_factor"),
-            py::arg("memory_config") = std::nullopt});
+            py::arg("input_tensor"), py::arg("scale_factor"), py::arg("memory_config") = std::nullopt});
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/data_movement_pybind.hpp:42]: missed it before I guess. Memory_config should be a kw only arg

File: ttnn/cpp/ttnn/operations/data_movement/tilize/tilize_pybind.hpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            [](const OperationType &self,
+               const ttnn::Tensor &input_tensor,
+               const std::optional<MemoryConfig> &memory_config,
+               std::optional<DataType> output_dtype,
+               bool use_multicore,
+               uint8_t queue_id) { return self(queue_id, input_tensor, memory_config, output_dtype, use_multicore); },
+            py::arg("input_tensor"),
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("output_dtype") = std::nullopt,
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/tilize/tilize_pybind.hpp:52]: for consistency with other ops
```suggestion
            py::arg("dtype") = std::nullopt,
```

File: ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding_pybind.hpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+               uint8_t queue_id) {
+                return self(
+                    queue_id, input_tensor, output_tensor_shape, value, memory_config, output_dtype, use_multicore);
+            },
+            py::arg("input_tensor"),
+            py::arg("output_tensor_shape"),
+            py::arg("pad_value"),
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("output_dtype") = std::nullopt,
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding_pybind.hpp:61]: dtype

File: ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding_pybind.hpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            [](const OperationType &self,
+               const ttnn::Tensor &input_tensor,
+               const std::optional<MemoryConfig> &memory_config,
+               std::optional<DataType> output_dtype,
+               bool use_multicore,
+               uint8_t queue_id) { return self(queue_id, input_tensor, memory_config, output_dtype, use_multicore); },
+            py::arg("input_tensor"),
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("output_dtype") = std::nullopt,
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding_pybind.hpp:103]: dtype

**Review by johanna-rock-tt — APPROVED** [review]
Falcon40b changes look good

**Review by skhorasganiTT — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Review by nsmithtt — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by AleksKnezevic — APPROVED** [review]

**Pull request merged**